README.rst

```diff
@@ -1723,6 +1723,18 @@ Choose from the list of available rules:
   - ``scalar_types`` (``bool``): fix also scalar types; may have unexpected
     behaviour due to PHP bad type coercion system; defaults to ``true``
 
+* **phpdoc_to_property_type**
+
+  EXPERIMENTAL: Takes ``@var`` annotation of non-mixed types and adjusts
+  accordingly the property signature. Requires PHP >= 7.4.
+
+  *Risky rule: [1] This rule is EXPERIMENTAL and is not covered with backward compatibility promise. [2] ``@var`` annotation is mandatory for the fixer to make changes, signatures of properties without it (no docblock, inheritdocs) will not be fixed. [3] Manual actions are required if inherited signatures are not properly documented. [4] ``@inheritdocs`` support is under construction.*
```

Comment:
We should also mention that nullable properties may be manually initialized if they are accessed before initialization.
The following code would indeed break with this fixer:
```php
class Foo
{
    /** @var null|Bar */
    private $bar;
    public function isBarGood(): bool
    {
        return null !== $this->bar;
    }
}
```
If you'd like to initialize nullable properties directly within this very fixer, I'd suggest to make it an option defaulted to `false`, because stricter code should always be favored.

Reply:
I like your initialization option idea but I'd leave it to a subsequent PR so we can move forward with this one. I'll update the risky description :+1:

Reply:
I don't think we've to document the way PHP works but limit us to what the fixer does and the docs go in enough detail IMHO.
The suggestion for adding setting the null-value within this fixer is a +0 for me and could be added later if wanted, so for me out of scope of the PR. Note that when the property is initialized already in the constructor it should not be added to the property declaration for example.

Reply:
To both: regarding the option/new-fixer implementation, of course this is not the place, it's much better to land this as it is right now, and think about it later.

> I don't think we've to document the way PHP works but limit us to what the fixer does and the docs go in enough detail IMHO.

I disagree: prior 7.4 no one ever initialized nullable properties to null, and now all of a sudden this fixer breaks everything even though apparently should be harmless.

And what this fixer does is changing the behavior of the code.

```php
    /** @var null|Bar */
    private $bar;
```

Means the property may be null or Bar **and** is initialized to null.

```php
    private ?Bar $bar;
```

This instead solely means may be null or Bar, but uninitialized.

> Note that when the property is initialized already in the constructor it should not be added to the property declaration for example.

This is technically valid also outside constructor.

Reply:
>though apparently should be harmless

I think this where we disagree on, besides being marked as risky the whole description already yells twice that it is an experimental fixer, I don't understand how people could read it as being harmless.

Reply:
The risky description already list all the downside known, you either delete them all and leave just a risky label, or put them all, I don't understand why pinpointing only half of them :confused:

Reply:
Description updated.